File: Spring Boot/microservices/place-service/src/main/java/pl/edu/pg/benchmarking/place/PlaceController.java
package pl.edu.pg.benchmarking.place;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestParam;
import pl.edu.pg.benchmarking.place.dto.GetPlaceResponse;
import pl.edu.pg.benchmarking.place.dto.GetPlacesAllResponse;
import pl.edu.pg.benchmarking.place.dto.GetPlacesResponse;

import java.util.List;
import java.util.Optional;
import java.util.Set;

@RestController
@RequestMapping("api/places")
public class PlaceController {

    private final PlaceRepository placeRepository;

    @Autowired
    public PlaceController(PlaceRepository placeRepository) {
        this.placeRepository = placeRepository;
    }

    @GetMapping("/all")
    public ResponseEntity<Iterable<GetPlacesAllResponse.Place>> getAllPlaces() {
        List<Place> resultList = placeRepository.findAll();
        return ResponseEntity.ok(
                GetPlacesAllResponse.entityToDtoMapper().apply(resultList));
    }

    @GetMapping("{id}")
    public ResponseEntity<GetPlaceResponse> getSinglePlace(@PathVariable(name = "id") Integer id) {
        Optional<Place> placeOptional = placeRepository.findById(id);
        return placeOptional
                .map(place -> ResponseEntity.ok(GetPlaceResponse.entityToDtoMapper().apply(place)))
                .orElseGet(() -> ResponseEntity.noContent().build());
    }

    @GetMapping
    public ResponseEntity<Iterable<GetPlacesResponse.Place>> getPlaces(@RequestParam Set<Integer> ids) {
        List<Place> resultList = placeRepository.findByIds(ids);
        return ResponseEntity.ok(
                GetPlacesResponse.entityToDtoMapper().apply(resultList));
    }

}

File: Flask/microservices/place-service/website/place_controller.py
from website.place_repository import PlaceRepository
import website.dto.get_place_response as place_response
import website.dto.get_places_response as places_response
from flask import Blueprint, jsonify, request
from static.utils import parse_str_to_list


place_views = Blueprint('place', __name__)
place_repository = PlaceRepository()


@place_views.route('/all', methods=['GET'])
def get_all_places() -> dict:
    places = place_repository.find_all()
    dto = places_response.entity_to_dto_mapper(places)
    return jsonify(dto)


@place_views.route('/<int:place_id>', methods=['GET'])
def get_single_place(place_id: int) -> dict:
    place = place_repository.find_by_id(place_id)
    if not place:
        return "Bad request - Place of given id does not exist", 404
    dto = place_response.entity_to_dto_mapper(place)
    return jsonify(dto.as_dict())

@place_views.route('', methods=['GET'])
def get_places() -> dict:
    ids_param = parse_str_to_list(request.args["ids"])
    result_list = place_repository.find_by_ids(ids_param)
    dto = places_response.entity_to_dto_mapper(result_list)
    return jsonify(dto)
